src/nodes/QskArcNode.cpp

```diff
@@ -89,29 +100,44 @@ void QskArcNode::setArcData( const QRectF& rect, const QskArcMetrics& arcMetrics
     const auto arcRect = qskEffectiveRect( rect, borderWidth );
     if ( arcRect.isEmpty() )
     {
+        delete shadowNode;
         delete fillNode;
         delete borderNode;
-
         return;
     }
 
     const auto path = metrics.painterPath( arcRect );
 
+    if ( shadowColor.alpha() > 0.0 )
```

Comment:
@uwe One could argue a if the gradient and stroke aren't visible the shadow is also not visible